website.hits

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh -eu
+
+readonly CURSOR_FILE=data/websites-cursor
+
+cursor=`cat $CURSOR_FILE || echo 0`
+
+new_lines=$(ssh traefik@srv1.c4dt.org tail -n +$cursor /var/log/traefik/access)
```

Comment:
Do I understand correctly that this automatically fetches all hosts from the `traefik/access` log? To me this gives a list that is very chaotic and on the verge of unusable...

I'd like to have all the demos, the different services we have (showcase, stats, cryptpad, matrix), and the `login.c4dt.org`. 

Bonus for 
- including `c4dt.org` in the list
- counting the same IP only once per time-interval

Reply:
Can't you include the apache logs? That should give you enough information as to which site is being requested.

Reply:
Or perhaps you can get the `RequestAddress` as described here:

https://doc.traefik.io/traefik/observability/access-logs/#limiting-the-fieldsincluding-headers

Under _available fields_.

Reply:
> Can't you include the apache logs? That should give you enough information as to which site is being requested.

Apache doesn't help much
```
172.20.0.4 - - [19/Jan/2022:21:33:29 +0000] "-" 408 0 "-" "-"
172.20.0.4 - - [19/Jan/2022:21:33:29 +0000] "-" 408 0 "-" "-"
172.20.0.1 - - [20/Jan/2022:16:33:20 +0000] "GET / HTTP/1.1" 200 10956 "-" "curl/7.58.0"
172.20.0.1 - - [20/Jan/2022:16:33:40 +0000] "GET /assets/css/main.css HTTP/1.1" 404 433 "-" "curl/7.58.0"
172.20.0.1 - - [20/Jan/2022:16:33:49 +0000] "GET /assets/css/main.css HTTP/1.1" 404 433 "-" "curl/7.58.0"
```

> Or perhaps you can get the `RequestAddress`

I tried to enable all fields, to see how the log lines changes, but it doesn't. Maybe because I'm using the default (CLF) format, I'll try with JSON.

Reply:
Soo, it seems that I've way way more information with JSON, going for that.

Reply:
OK - from what I understand in the link above, you can also add/remove fields in the traefik log. Same thing for apache:

https://httpd.apache.org/docs/2.4/logs.html

Where it describes how you can define what will be written to the log files.